src/Model.php

```diff
@@ -620,7 +630,8 @@ public function isDirty(string $field): bool
     {
         $this->checkOnlyFieldsField($field);
 
-        if (array_key_exists($field, $this->dirty)) {
+        $dirtyRef = &$this->getDirtyRef();
+        if (array_key_exists($field, $dirtyRef)) {
```

Comment:
And this method looks exaactly the same as `_isset($field)` method below, so to avoid code duplication we could just use `return $this->_isset($field);` here.